importer/import.go

```diff
@@ -0,0 +1,282 @@
+	Value       float64
+	Labels      tsdb_labels.Labels
+}
+
+// ImportFromFile imports data from a file formatted according to the Prometheus exposition format,
+// converts it into a snapshot/block, and places the newly created snapshot/block in the
+// TSDB DB directory, where it is treated like any other block.
+func ImportFromFile(filePath string, contentType string, dbPath string, skipTimestampCheck bool, logger log.Logger) error {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+
+	f, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	bytes, err := ioutil.ReadAll(f)
+	if err != nil {
+		return err
+	}
+
+	metricSamples, minValidTimestamp, maxValidTimestamp, err := parseMetrics(bytes, contentType)
+	if err != nil {
+		return err
+	}
+
+	if !skipTimestampCheck {
+		err = verifyIntegration(dbPath, minValidTimestamp, maxValidTimestamp)
+		if err != nil {
+			return err
+		}
+	}
+
+	tmpDbDir, err := ioutil.TempDir("", "importer")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDbDir)
+
+	//snapshotPath, err := pushToDisk(metricSamples, tmpDbDir, minValidTimestamp, logger)
+	snapshotPaths, err := pushToDisk(metricSamples, tmpDbDir, logger)
+	if err != nil {
+		return err
+	}
+	level.Info(logger).Log("msg", "blocks created", "blockPaths", snapshotPaths)
+
+	err = copyToDatabase(tmpDbDir, dbPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// parseMetrics parses metrics formatted in the Prometheus exposition format.
+// Returns the metric samples, min timestamp, max timestamp, and error.
+func parseMetrics(b []byte, contentType string) ([][]*metricSample, timestamp, timestamp, error) {
+	var minValidTimestamp timestamp
+	minValidTimestamp = math.MaxInt64
+	var maxValidTimestamp timestamp
+	maxValidTimestamp = math.MinInt64
+
+	var buckets [][]*metricSample
+	var currentBucket []*metricSample
+	var startTime time.Time
+	var err error
+	parser := textparse.New(b, contentType)
+	for {
+		var ent textparse.Entry
+		if ent, err = parser.Next(); err != nil {
+			// Error strings are just differently enough across packages,
+			// hence this catch-all that just looks for "EOF" in the error
+			// string, and if it finds one, it means that the parsing is complete.
+			if strings.Contains(strings.ToLower(err.Error()), "eof") {
+				err = nil
+				break
+			}
+			// In case the error that we see is not related to the EOF.
+			return nil, minValidTimestamp, maxValidTimestamp, err
+		}
+		switch ent {
+		case textparse.EntryType:
+			continue
+		case textparse.EntryHelp:
+			continue
+		case textparse.EntryUnit:
+			continue
+		case textparse.EntryComment:
+			continue
+		default:
+		}
+		_, currentTimestampNs, val := parser.Series()
+
+		// Parses converts all timestamps to nanoseconds.
+		// TSDB looks for timestamps in milliseconds.
+		var currentTimestampMs timestamp
+		if currentTimestampNs == nil {
+			currentTimestampMs = 0
+		} else {
+			currentTimestampMs = *currentTimestampNs / 1000
```

Comment:
That's not the right factor

Reply:
Yes, you're right. The parser returns values in microseconds actually.